apps/sertra/pages/server/[id].tsx

```diff
@@ -30,8 +30,34 @@ export const getStaticProps: GetStaticProps<
   // is named appropriately.
   const listingId = context.params!.id; // eslint-disable-line @typescript-eslint/no-non-null-assertion
   const res = await fetch(ApiURLs.ServerDetail(listingId));
+  const mods_res = await fetch(TsApiURLs.V1Packages("v-rising"));
   const data = await res.json();
+  const mods_data = await mods_res.json();
 
+  const updated_listing_mods_data = [];
+  for (const mod_ref of data.mods) {
+    for (const mod of mods_data) {
+      const [mod_ref_owner, mod_ref_name, mod_ref_version] = mod_ref.split(
+        "-",
+        3
+      );
+      if (mod_ref_owner === mod.owner && mod_ref_name === mod.name) {
+        for (const version of mod.versions) {
+          if (mod_ref_version === version.version_number) {
+            updated_listing_mods_data.push({
+              name: mod.name,
+              owner: mod.owner,
+              version: version.version_number,
+              icon_url: version.icon,
+              description: version.description ?? null,
+            });
+          }
+        }
+      }
+    }
+  }
+  data.mods = updated_listing_mods_data;
```

Comment:
I think mutating the `data`, which is a fetch response(?) is a bad idea. The processed data should be stored in state separately.

Reply:
Good point, dunno why I didn't just plug the new array into the return :D Fixd, this 👍